lib/data/wrapper.js

```diff
@@ -59,6 +59,23 @@ if (config.backends.data === 'mem') {
  */
 const MAX_RETRY = 2;
 
+// This check is done because on a put, complete mpu or copy request to AWS,
+// if the object already exists on AWS, the existing object should not be
+// deleted, which is the functionality for all other backends
+function _shouldSkipDelete(locations, requestMethod, newObjDataStoreName) {
+    const skipBackend = externalBackends;
+    const skipMethods = { PUT: true, POST: true };
+    if (!Array.isArray(locations) || !locations[0] ||
+        !locations[0].dataStoreType) {
+        return false;
```

Comment:
why do you have to still delete if there is no data?

Reply:
The batch delete uses an async.eachLimit method so it won't do anything for an empty array even if we return `false`.
I return early if it's an empty array here so we won't get an error below accessing the `dataStoreType` and `dataStoreName` properties of the 0th index.